pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -819,7 +819,7 @@ def get_hamiltonian(self, time: float) -> qutip.Qobj:
     def run(
         self,
         progress_bar: Optional[bool] = False,
-        **options: qutip.solver.Options,
+        **options: dict[str, Any],
```

Comment:
Actually, `**kwargs` are type-hinted by specified the type hint of the values (which requires that all values are of the same type). [PEP 692](https://peps.python.org/pep-0692/) actually aims to improve this, but for now I'm afraid we'll have to fall back on doing it like this:
```suggestion
        **options: Any
```